extensions/amp-analytics/0.1/test/test-cookie-writer.js

```diff
@@ -292,6 +292,139 @@ describes.fakeWin('amp-analytics.cookie-writer value', {amp: true}, env => {
     });
   });
 
+  it('should write cookie with custom expiration (number value)', () => {
+    win.location = 'https://www.example.com/';
+    const cookieWriter = new CookieWriter(
+      win,
+      win.document.body,
+      dict({
+        'cookies': {
+          'cookieMaxAge': 604800, // 1 week in seconds
+          'aCookie': {
+            'value': 'testValue',
+          },
+        },
+      })
+    );
+    return cookieWriter.write().then(() => {
+      expect(win.document.lastSetCookieRaw).to.equal(
+        'aCookie=testValue; path=/; domain=example.com; ' +
+          'expires=Mon, 08 Jan 2018 08:00:00 GMT'
+      );
+    });
+  });
+
+  it('should write cookie with custom expiration (string value)', () => {
+    win.location = 'https://www.example.com/';
+    const cookieWriter = new CookieWriter(
+      win,
+      win.document.body,
+      dict({
+        'cookies': {
+          'cookieMaxAge': '604800', // 1 week in seconds
+          'aCookie': {
+            'value': 'testValue',
+          },
+        },
+      })
+    );
+    return cookieWriter.write().then(() => {
+      expect(win.document.lastSetCookieRaw).to.equal(
+        'aCookie=testValue; path=/; domain=example.com; ' +
+          'expires=Mon, 08 Jan 2018 08:00:00 GMT'
+      );
+    });
+  });
+
+  it('should write cookie with custom expiration (decimal value)', () => {
+    win.location = 'https://www.example.com/';
+    const cookieWriter = new CookieWriter(
+      win,
+      win.document.body,
+      dict({
+        'cookies': {
+          'cookieMaxAge': 604800.000123,
+          'aCookie': {
+            'value': 'testValue',
+          },
+        },
+      })
+    );
+
+    return cookieWriter.write().then(() => {
+      expect(win.document.lastSetCookieRaw).to.equal(
+        'aCookie=testValue; path=/; domain=example.com; ' +
+          'expires=Mon, 08 Jan 2018 08:00:00 GMT'
+      );
+    });
+  });
+
+  it('should write cookie with default expiration (invalid string value)', () => {
+    win.location = 'https://www.example.com/';
+    const cookieWriter = new CookieWriter(
+      win,
+      win.document.body,
+      dict({
+        'cookies': {
+          'cookieMaxAge': 'invalid',
```

Comment:
nit: May want to assert `allowConsoleError`or `expectAsyncConsoleError` functions to examine user error type.